app/models/upload.rb

```diff
@@ -546,6 +510,21 @@ def self.migrate_to_new_scheme(limit: nil)
     problems
   end
 
+  def self.extract_upload_ids(raw)
+    return [] if raw.blank?
+
+    sha1s = []
+
+    raw.scan(/upload:\/\/([a-zA-Z0-9]+)(?:\.[a-zA-Z0-9]+)?/).each do |match|
```

Comment:
Why not use the same REGEXP as in https://github.com/discourse/discourse/blob/b96f90f9705a8f62222939fe3e613a215fdb2538/app/models/upload.rb#L337-L341 ?

Reply:
There is a difference between the two regular expressions. The regular expression you propose is ideal when the whole string is a short URL. The regular expression I use is looking explicitly for `upload://`. I could probably make it shorter doing similar to that.